### PR TITLE
[Merged by Bors] - Add LCH(ab) color space to `bevy_render::color::Color`

### LINKED ISSUE #766: LCH Color Function
**What problem does this solve or what need does it fill?**

Colors are difficult to read. For example, it's hard to tell that `Color::rgb(0.8, 0.3, 0.7)` is pink.

**Describe the solution would you like?**

Add a LCH color function as well as the HEX colors. The LCH color space allows for colors that are somewhat related to how people actually see it.

**Describe the alternative(s) you've considered?**

N/A

**Additional context**

This blog post (https://lea.verou.me/2020/04/lch-colors-in-css-what-why-and-how/) provides a good explanation of what the LCH color space is.

### PR DESCRIPTION
**Pull request by ldubos**
# Objective

- Fixes #766

## Solution

- Add a new `Lcha` member to `bevy_render::color::Color` enum

---

## Changelog

- Add a new `Lcha` member to `bevy_render::color::Color` enum
- Add `bevy_render::color::LchRepresentation` struct

### CODE REVIEW

File: crates/bevy_render/src/color/colorspace.rs [hunk 1]
```diff
@@ -102,6 +102,123 @@ impl HslRepresentation {
     }
 }
 
+pub struct LchRepresentation;
```
Comment (alice-i-cecile) [inline crates/bevy_render/src/color/colorspace.rs:105]: Doc strings please. Links to reference materials would be particularly useful.

File: crates/bevy_render/src/color/colorspace.rs [hunk 2]
```diff
@@ -102,6 +102,123 @@ impl HslRepresentation {
...
+impl LchRepresentation {
+    const CIE_EPSILON: f32 = 216.0 / 24389.0;
```
Comment (alice-i-cecile) [inline crates/bevy_render/src/color/colorspace.rs:107]: All of these magic numbers need to be sourced. A single good reference is fine, but ideal would be specific callouts.

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
Well-made and tested, but needs more documentation and references :)

**Review by alice-i-cecile — APPROVED** [review]
Thanks :)

**Comment by alice-i-cecile** [comment]
@iMplode-nZ, care to review?

**Review by mockersf — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
bors retry

**Pull request closed without merging**